blocks/monitoring/__init__.py

```diff
@@ -0,0 +1,193 @@
+"""This part of the framework helps you monitor your model during training.
+
+:class:`MonitorChannels` describe quantities that should be monitored. They
+can either be Theano variables (e.g. the objective function, the weight
+norms, etc.) or they can be callable functions (e.g. if you want to use
+NumPy to calculate eigenvalues, or sample text from your language model).
+
+.. todo::
+
+   Monitored Theano variables might depend on inputs beyond those given to
+   the model. There should be a way of providing these to the monitoring
+   channels, collecting them, and compiling them as part of the Theano
+   function that performs monitoring.
+
+"""
+import logging
+from abc import ABCMeta, abstractmethod
+
+from blocks.utils import shared_expression, update_instance
+
+logger = logging.getLogger(__name__)
+
+
+class MonitorChannel(object):
+    """A channel to monitor
+
+    Parameters
+    ----------
+    name : str
+        The name of this monitor channel, should be unique in order to
+        distinguish channels.
+
+    """
+    def __init__(self, name):
+        update_instance(self, locals())
+
+
+class VariableMonitorChannel(MonitorChannel):
```

Comment:
Because it depended on the `name` property of the `expression` attribute, but the `expression` attribute didn't necessarily exist (it was added manually by the `aggregated_div` function), so it would have crashed if you instantiated the `AggregatedDiv` class yourself.